File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int n = nums.size();
        long long int pre[n];
        for(int i = 0 ;i <n;i++){
            pre[i] = nums[i];
            if(i) pre[i] += pre[i-1];
        }
        int cnt = 0;
        for(int i = 0 ;i <n-1;i++){
            if(pre[i] >= pre[n-1] - pre[i]) cnt++;
        }
        return cnt;
    }
};

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.cs
public class Solution {
    public int WaysToSplitArray(int[] nums) {
        int n = nums.Length, ValidSplits = 0;
        long [] prefixSum = new long [n];
        
        for(int i = 0 ; i < n; i++){
            prefixSum[i] = nums[i];
            if(i > 0) prefixSum[i] += prefixSum[i-1];
        }
        for(int i = 0 ; i< n-1;i++){
            if(prefixSum[i] >= prefixSum[n-1] - prefixSum[i]) ValidSplits++;
        }
        return ValidSplits;
    }
}